From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Fri, 30 Sep 2016 11:18:55 +0200
Subject: Reorder definitions to make omitting tests easier

Cherry-picking commit e4b37f7.

Conflicts:
	Makefile.am
	lrmd/Makefile.am
	pengine/Makefile.am

Gbp-Pq: Topic from-upstream
---
 Makefile.am         | 4 ++--
 lrmd/Makefile.am    | 6 +++---
 pengine/Makefile.am | 9 ++++-----
 3 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 0edf0c9..7edb6d5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -27,7 +27,6 @@ CORE	= $(LIBLTDL_DIR) replace include lib mcp attrd pengine cib crmd fencing lrm
 SUBDIRS	= $(CORE) cts extra doc
 
 doc_DATA = AUTHORS COPYING COPYING.LIB
-noinst_PROGRAMS = scratch
 
 ACLOCAL_AMFLAGS  = -I m4
 
@@ -36,7 +35,8 @@ test_SCRIPTS		= coverage.sh BasicSanity.sh
 test_DATA		= valgrind-pcmk.suppressions
 
 # Scratch file for ad-hoc testing
-scratch_SOURCES	= scratch.c
+noinst_PROGRAMS = scratch
+nodist_scratch_SOURCES	= scratch.c
 scratch_LDADD	= $(top_builddir)/lib/common/libcrmcommon.la -lm
 
 core:
diff --git a/lrmd/Makefile.am b/lrmd/Makefile.am
index a8fb07a..c20ecd9 100644
--- a/lrmd/Makefile.am
+++ b/lrmd/Makefile.am
@@ -17,12 +17,12 @@
 
 include $(top_srcdir)/Makefile.common
 
+lrmdlibdir	= $(CRM_DAEMON_DIR)
+lrmdlib_PROGRAMS	= lrmd lrmd_test lrmd_internal_ctl
+
 testdir			= $(datadir)/$(PACKAGE)/tests/lrmd
 test_SCRIPTS		= regression.py
 
-lrmdlibdir	= $(CRM_DAEMON_DIR)
-lrmdlib_PROGRAMS = lrmd lrmd_test lrmd_internal_ctl
-
 initdir		 = $(INITDIR)
 init_SCRIPTS	 = pacemaker_remote
 sbin_PROGRAMS	 = pacemaker_remoted
diff --git a/pengine/Makefile.am b/pengine/Makefile.am
index 4bf82dd..871e2ff 100644
--- a/pengine/Makefile.am
+++ b/pengine/Makefile.am
@@ -30,11 +30,6 @@ test_DATA		= regression.core.sh
 test10dir		= $(datadir)/$(PACKAGE)/tests/pengine/test10
 test10_DATA		= $(PE_TESTS) $(PE_TESTS:%.scores=%.xml) $(PE_TESTS:%.scores=%.exp) $(PE_TESTS:%.scores=%.dot) $(PE_TESTS:%.scores=%.summary) $(wildcard test10/*.stderr)
 
-COMMONLIBS	= 					\
-		$(top_builddir)/lib/common/libcrmcommon.la		\
-		$(top_builddir)/lib/pengine/libpe_status.la		\
-		libpengine.la $(CURSESLIBS) $(CLUSTERLIBS)
-
 beekhof:
 	echo $(shell ls -1 test10/*.xml)
 
@@ -46,6 +41,10 @@ AM_XML_LOG_FLAGS = -V --run
 #LOG_COMPILER = 
 #AM_LOG_FLAGS = -V
 
+COMMONLIBS	= $(top_builddir)/lib/common/libcrmcommon.la	\
+		$(top_builddir)/lib/pengine/libpe_status.la	\
+		libpengine.la
+
 
 ## libraries
 lib_LTLIBRARIES	= libpengine.la
